operator/internal/lifecycle/v2_simple_resources.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package lifecycle
+
+import (
+	"context"
+
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/redpanda-data/redpanda-operator/charts/redpanda/v5"
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha2"
+	"github.com/redpanda-data/redpanda-operator/pkg/gotohelm/helmette"
+	"github.com/redpanda-data/redpanda-operator/pkg/kube"
+)
+
+// V2SimpleResourceRenderer represents an simple resource renderer for v2 clusters.
+type V2SimpleResourceRenderer struct {
+	kubeConfig clientcmdapi.Config
```

Comment:
This will need to be a `rest.Config` soon, if it doesn't already. You may be racing with Rafal's update PR. Not too difficult to fix in a conflict thankfully.